### PR TITLE
Feat/Create Events and Enhancements

### PR DESCRIPTION
**Pull request by johnnabil**
##### Why do we need this change?
This PR is for Creating the Events feature and make small enhancements based on the mobile team needs
- developing the events table CRUD operations
- return all events in one function
- returning IDs with findMany functions
- retrieving the profile of the currently logged-in user

### CODE REVIEW

File: src/modules/Event/repos/eventRepo.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            ${take} 
+        OFFSET 
+            ${skip}
+    `;
+
+    return {
+      Birthdays,
+      Anniversaries,
+      Events,
+    };
```
Comment (kerolloz) [inline src/modules/Event/repos/eventRepo.ts:127]: naming, my friend

Comment (johnnabil) [reply src/modules/Event/repos/eventRepo.ts:127]: What is the problem?

Comment (kerolloz) [reply src/modules/Event/repos/eventRepo.ts:127]: Why pascal case?

File: src/database/prisma/schema.prisma [hunk 2]
```diff
@@ -38,8 +38,10 @@ model User {
   nominatedUser     KlivvrPickNominee[] @relation("Nominee")
   nominatorUser     KlivvrPickNominee[] @relation("Nominator")
   Like              Like[]
+  events            Event[]
   Notification      Notification[]
   DeviceToken       DeviceToken?
+  events            Event[]
```
Comment (mohamadelhinamy) [inline src/database/prisma/schema.prisma:44]: why is there 2?

File: src/modules/Event/repos/eventRepo.ts [hunk 3]
```diff
@@ -28,103 +28,28 @@ export class EventRepo {
...
-            date_part('month', "hiringDate") = date_part('month', current_date)
+            date_part('month', "date") = date_part('month', current_date)
             AND
-            date_part('day', "hiringDate") = date_part('day', current_date)
+            date_part('day', "date") = date_part('day', current_date)
             OR 
-            date(date_part('year', current_date)||'-'||date_part('month', "hiringDate")||'-'||date_part('day', "hiringDate")) BETWEEN current_date AND current_date + interval '7 days'
-        Order by
-            "inKlivvrSince" DESC
+            date(date_part('year', current_date)||'-'||date_part('month', "date")||'-'||date_part('day', "date")) BETWEEN current_date AND current_date + interval '7 days'
```
Comment (kerolloz) [inline src/modules/Event/repos/eventRepo.ts]: stay consistent `DATE_PART` or `date_part`, I suggest following the SQL convention of using all caps

File: src/database/prisma/schema.prisma (hunk 2 above)
Comment (johnnabil) [reply src/database/prisma/schema.prisma]: sorry kan 3andena git crisis benhelaha 😂

File: src/modules/Event/services/eventService.ts [hunk 4]
```diff
@@ -9,8 +9,13 @@ export class EventService {
   readonly createEventSchema = z.object({
     name: z.string().trim().nonempty(),
     date: z.coerce.date(),
+<<<<<<< HEAD
```
Comment (kerolloz) [inline src/modules/Event/services/eventService.ts]: conflicts

File: src/modules/Event/services/eventService.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import { CustomError } from '../../../middlewares';
+
+export class EventService {
+  constructor(private readonly eventRepo: EventRepo) {}
+
+  readonly createEventSchema = z.object({
+    name: z.string().trim().nonempty(),
+    date: z.coerce.date(),
+    startTime: z.coerce.date().optional(),
+    endTime: z.coerce.date().optional(),
```
Comment (mohamadelhinamy) [inline src/modules/Event/services/eventService.ts:13]: arent the startTime and endTime required in the database why are they optional here?

**Review by msherifs — APPROVED** [review]

File: src/routes/eventRouter.ts [hunk 6]
```diff
@@ -1,23 +1,91 @@
...
+);
 
-    res.status(200).json({ usersWithBirthdays, usersAnniversaries });
+router.delete(
+  '/:id',
+  endpoint(async (req, res) => {
+    const id = Number(req.params.id);
+    await eventService.findOneWithError({ id, userId: req.user?.id });
+    const deletedKlivvrPickObject = await eventService.deleteOne({ id });
+    res.status(200).json(deletedKlivvrPickObject);
```
Comment (kerolloz) [inline src/routes/eventRouter.ts:88]: `deletedKlivvrPickObject`?

**Review by msherifs — APPROVED** [review]

**Pull request merged**